repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/CaseFieldEntityUtil.java

```diff
@@ -12,19 +9,27 @@
 import static org.apache.commons.lang3.StringUtils.isBlank;
 
 import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.ComplexFieldEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.FieldEntity;
 
 @Component
 public class CaseFieldEntityUtil {
 
-    public CaseFieldEntityUtil() { }
-
     public List<String> buildDottedComplexFieldPossibilities(List<? extends FieldEntity> caseFieldEntities) {
+        return removeElementsThatAreNotTreeLeafs(buildAllDottedComplexFieldPossibilities(caseFieldEntities));
+    }
+
+    public List<String> buildDottedComplexFieldPossibilitiesIncludingParentComplexFields(List<? extends FieldEntity> caseFieldEntities) {
+        return removeElementsThatAreCaseFields(buildAllDottedComplexFieldPossibilities(caseFieldEntities), caseFieldEntities);
+    }
+
+    private List<String> buildAllDottedComplexFieldPossibilities(List<? extends FieldEntity> caseFieldEntities) {
         List<String> allSubTypePossibilities = new ArrayList<>();
         List<? extends FieldEntity> fieldEntities = caseFieldEntities.stream()
             .filter(Objects::nonNull)
             .collect(Collectors.<FieldEntity>toList());
         prepare(allSubTypePossibilities, "", fieldEntities);
-        return removeElementsThatAreNotTreeLeafs(allSubTypePossibilities);
+        return allSubTypePossibilities;
     }
 
     private List<String> removeElementsThatAreNotTreeLeafs(List<String> allSubTypePossibilities) {
```

Comment:
Spelling error.

Reply:
You mean `That -> Those`?

Reply:
`TreeLeafs` instead of `TreeLeaves` but it may have been there before 🤔

Reply:
We can leave it - up to you